packages/oidc-dpop-client-browser/src/dpop/tokenExchange.spec.ts

```diff
@@ -0,0 +1,430 @@
+/*
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import URL from "url-parse";
+import { IClient, IIssuerConfig } from "@inrupt/solid-client-authn-core";
```

Comment:
I think I've mentioned this before, but ideally this package does not depend on SCAC.

Reply:
Initially that's the approach I was trying (probably because you suggested it), but that resulted in the types `IClient` and `IIssuerConfig` being exported by this browser-specific module, and imported by the generic `core` module, which didn't feel right. I hesitated to create a small module that would just export these types, making `core` and `oidc-client-dpop-browser` independant, but that felt a bit overkill, even if I guess it would be the 'right' thing to do. What do you think ?

Reply:
Since it's an interface that describes the parameters you have to pass to oidc-dpop-client, that does feel right to me. If it so happens to be that SCAC accepts equal arguments in its own API, then I think it's fair to have it define its own interface as well, that just so happens to be equal to this one. That would also mean that you would be warned if you simply pass on arguments given to SCAC that are no longer compatible with oidc-dpop-client.

Reply:
Resolved in bf8dd3ebaaff79eecfe21bcd6849caad240879df